### PR TITLE
Protect create2 verification with an auth0 authentication

### PR DESCRIPTION
**Pull request by marcocastignoli**
* using auth0 I created an Application API + Frontend Application + user database.
* in order to access the /verify/create2 API, a jwt token is necessary. ~and the user calling the api must have the "verify:create2" permission (set on the Application API)~ (functionality commented for now)
* in order to send request programmatically the developer needs to create an account and then "login" via cli get the token and send the token
* in the ui the user needs to login, and then a new token is asked every request
* canceling tokens mean to:
  * delete the user
  * set the jwt token expiracy data very short (like 30 seconds)

## TODO
- [x] implement tests
- [ ] Add instructions in the docs

### CODE REVIEW

**Comment by kuzdogan** [comment]
I came across a bug. It is not related to this PR but we can fix it here.

Steps to reproduce: 
1. Add a contract and click "Verify create2 contract"
2. Input a valid address
3. Input an **invalid** salt such as `-`

The UI will throw

**Comment by kuzdogan** [comment]
Also having an error after clicking "Verify" 
image

image

Can you successfully verify or am I missing a config?

**Review by kuzdogan — COMMENTED** [review]
Just a small thing + the bugs I've encountered.

**Comment by marcocastignoli** [comment]
> Also having an error after clicking "Verify"

I've never seen that error

> I came across a bug. It is not related to this PR but we can fix it here.

I'll check this

File: src/server/controllers/verification/verification.common.ts [hunk 1]
```diff
@@ -385,21 +388,32 @@ export const verifyContractsInSession = async (
...
+    message: {
+      error: errorMessage,
+    },
+    keyGenerator: (req: Request) => req.auth?.payload.sub as string,
+  });
+
+export const apiCheckPermission = (permission: string, errorMessage: string) =>
+  function (req: Request, res: Response, next: NextFunction) {
+    if (!(req.auth?.payload?.permissions as string)?.includes(permission)) {
+      throw new BadRequestError(errorMessage);
```
[the remark this answers is not in the document]
Comment (marcocastignoli) [reply src/server/controllers/verification/verification.common.ts:416]: right!

**Pull request merged**